src/order/disjointed.lean

```diff
@@ -87,4 +87,12 @@ lemma Union_disjointed_of_mono (hf : monotone f) (n : ℕ) :
 subset.antisymm (bUnion_subset $ λ k hk, subset.trans disjointed_subset $ hf $ nat.lt_succ_iff.1 hk)
   subset_Union_disjointed
 
+lemma Union_inter_disjointed_eq {a : set α} (ha : a ⊆ ⋃ n, f n) :
+  (⋃ n, a ∩ disjointed f n) = a :=
+by rwa [← inter_Union, set.Union_disjointed, inter_eq_left_iff_subset]
+
+lemma pairwise_disjoint_on_inter_disjointed {a : set α} :
+  pairwise $ disjoint on (λ n, a ∩ disjointed f n) :=
+pairwise_disjoint_on_inter disjoint_disjointed
```

Comment:
These lemmas might be better as just intermediate `have` statements in your proofs - they seem pretty contrived, and the proofs are short.

Reply:
I took the liberty and removed `pairwise_disjoint_on_inter` as well since its a similar lemma with a short proof